lib/request-wrapper.ts

```diff
@@ -391,16 +474,52 @@ export class RequestWrapper {
     }
     this.raxConfig = RequestWrapper.getRaxConfig(this.axiosInstance, retryOptions);
     this.retryInterceptorId = rax.attach(this.axiosInstance);
+    logger.debug(
+      `Enabled retries; maxRetries=${this.raxConfig.retry}, maxRetryInterval=${this.raxConfig.maxRetryDelay}`
+    );
   }
 
   public disableRetries(): void {
     if (typeof this.retryInterceptorId === 'number') {
       rax.detach(this.retryInterceptorId, this.axiosInstance);
       delete this.retryInterceptorId;
       delete this.raxConfig;
+      logger.debug('Disabled retries');
     }
   }
 
+  /**
+   * Returns true iff the previously-failed request contained in "error" should be retried.
+   * @param error - an AxiosError instance that contains a previously-failed request
+   * @returns true iff the request should be retried
+   */
+  private static retryPolicy(error: AxiosError) {
+    if (logger.debug.enabled) {
+      const details = [];
+      if (error.response) {
+        const statusText = error.response.statusText || ``;
+        details.push(`status_code=${error.response.status} (${statusText})`);
+      }
+      if (error.config) {
+        if (error.config.method) {
+          details.push(`method=${error.config.method.toUpperCase()}`);
+        }
+        if (error.config.url) {
+          details.push(`url=${error.config.url}`);
+        }
+      }
+
+      logger.debug(`Considering retry attempt; ${details.join(', ')}`);
+    }
+
+    // Delegate to the default function defined by retry-axios.
+    const shouldRetry = rax.shouldRetryRequest(error);
```

Comment:
I chickened out and just made this function a wrapper around the default implementation found in retry-axios.  Later, if we decide to make the retry criteria 100% consistent across the cores, we could change this to perform our custom logic here instead.